app-emulation/playonlinux/files/playonlinux-4.2.11-fix-makefile.patch

```diff
@@ -0,0 +1,21 @@
+diff -Naur playonlinux-orig/Makefile playonlinux/Makefile
+--- playonlinux-orig/Makefile	2017-06-14 19:31:55.746343655 -0700
++++ playonlinux/Makefile	2017-06-14 19:35:22.616281108 -0700
+@@ -28,8 +28,6 @@
+ PREFIX ?= /usr
+ DESTDIR ?= # root dir
+ 
+-CFLAGS += -lGL -lX11
+-
+ sharedir := $(DESTDIR)$(PREFIX)/share
+ bindir := $(DESTDIR)$(PREFIX)/bin
+ execdir := $(DESTDIR)$(PREFIX)/libexec
+@@ -46,7 +44,7 @@
+ 	$(RM) ./ChangeLog
+ 
+ build:
+-	$(CC) ./src/check_direct_rendering.c -o ./bin/playonlinux-check_dd
++	$(CC) ./src/check_direct_rendering.c -o ./bin/playonlinux-check_dd -lGL -lX11
```

Comment:
Not so - if `CFLAGS` and `LDFLAGS` were excluded, then this bug would not occur. The only thing I changed was to add that `-lGL -lX11` on to the end of the line that invokes gcc. When I build the package with my patched Makefile gcc is invoked as:
```
gcc -O2 -pipe -march=haswell -Wl,-O1 -Wl,--as-needed -Wl,-z,now -Wl,-z,relro ./src/check_direct_rendering.c -o ./bin/playonlinux-check_dd -lGL -lX11
```
whereas with the upstream Makefile I get
```
gcc -O2 -pipe -march=haswell -Wl,-O1 -Wl,--as-needed -Wl,-z,now -Wl,-z,relro -lGL -lX11 ./src/check_direct_rendering.c -o ./bin/playonlinux-check_dd
/var/tmp/portage/app-emulation/playonlinux-4.2.11/temp/ccEoJYO5.o: In function `main':
check_direct_rendering.c:(.text.startup+0x28): undefined reference to `XOpenDisplay'
check_direct_rendering.c:(.text.startup+0xe4): undefined reference to `glXChooseVisual'
check_direct_rendering.c:(.text.startup+0x102): undefined reference to `glXCreateContext'
check_direct_rendering.c:(.text.startup+0x112): undefined reference to `glXIsDirect'
check_direct_rendering.c:(.text.startup+0x169): undefined reference to `XFree'
check_direct_rendering.c:(.text.startup+0x182): undefined reference to `glXChooseVisual'
```
Note that the only difference here is that I have moved `-lGL -lX11` to the end of the arguments.


The upstream Makefile sets `CC = gcc $(CFLAGS)` further up in the file, and adds `-lGL -lX11` to `CFLAGS` (which I moved to the `$(CC)` invocation since it was only called once), not `LDFLAGS`. The Makefile doesn't touch `LDFLAGS` at all so the user's are applied unchanged.

I mentioned in my PR on the POL Github that they should be respecting the user's choice of compiler instead of forcing gcc, but my goal with this patch was to make the change as minimal as possible so as to fix the problem without breaking anything. My hope was that someone upstream with more knowledge of the codebase would fix the Makefile more broadly and that in the meantime this simple fix would solve matters.